code/api/src/modules/user/query.test.js

```diff
@@ -0,0 +1,77 @@
+import request from 'supertest';
+import express from 'express';
+import graphqlHTTP from 'express-graphql';
+import schema from '../../setup/schema';
+
+const app = express();
+
+describe("user queries", () => {
+  let server;
+
+  beforeAll(() => {
+    server = express();
+    server.use(
+      '/',
+      graphqlHTTP({
+        schema: schema,
+        graphiql: false
+      })
+    );
+  });
+
+  it('works', () => {
+    expect(true).toBe(true);
+  })
+
+  // It returns all users
+  it('returns all users', async () => {
+    const response = await request(server)
+      .post('/')
+      .send({ query: '{ users { name email } }' })
+      .expect(200)
+
+      expect(Object.keys(response.body.data)).toEqual(['users'])
```

Comment:
As a front-end person, this makes total sense to me, which is awesome! Thanks for the thorough explanation too